### PR TITLE
Adds babel/register to make nested components work

### LINKED ISSUE #1: Can't include other components
I'm trying to import components defined in the theme to use in a page, but it seems that is not intended to work since the dependencies don't get run through babel so the compiler is choking on the non-transpiled es6 syntax. Is that correct? What is the appropriate way to include theme components in a page?

### PR DESCRIPTION
**Pull request by claydiffrient**
This also bumps the babel-core version to the latest version as
well.

fixes #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 var babel = require('babel-core')
 var React = require('react')
 var reval = require('eval')
+require('babel/register')
```
Comment (vanwagonet) [inline index.js]: `register` is also in `babel-core`. Can you change this to `babel-core/register`?

File: package.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
   },
   "homepage": "https://github.com/thetalecrafter/hexo-renderer-react",
   "dependencies": {
-    "babel-core": "^5.5.8",
+    "babel": "^5.8.23",
```
Comment (vanwagonet) [inline package.json]: `babel` package shouldn't be necessary since we don't use any cli tools.

**Comment by claydiffrient** [comment]
@thetalecrafter Changes made, should work just fine now.

**Comment by vanwagonet** [comment]
You are a gentleman and a scholar @claydiffrient .

**Pull request merged**